src/main/java/network/brightspots/rcv/ClearBallotCvrReader.java

```diff
@@ -0,0 +1,144 @@
+  // see Clear Ballot 2.1 RCV Format Specification for details
+void readCastVoteRecords(List<CastVoteRecord> castVoteRecords) throws CvrParseException {
+    BufferedReader csvReader;
+    try {
+      csvReader = new BufferedReader(new FileReader(this.cvrPath));
+      // each "choice column" in the input Csv corresponds to a particular ranking: candidate+rank pair
+      // we parse these rankings from the header row into a map for lookup during CVR parsing
+      String firstRow = csvReader.readLine();
+      if (firstRow == null) {
+        Logger.log(Level.SEVERE, "No header row found in cast vote record file: %s", this.cvrPath);
+        throw new CvrParseException();
+      }
+      String[] headerData = firstRow.split(",");
+      if (headerData.length < CvrColumnField.ChoicesBegin.ordinal()) {
+        Logger.log(Level.SEVERE, "No choice columns found in cast vote record file: %s", this.cvrPath);
+      }
+      Map<Integer, Pair<Integer, String>> columnIndexToRanking = new HashMap<>();
+      for (int columnIndex = CvrColumnField.ChoicesBegin.ordinal(); columnIndex < headerData.length; columnIndex++) {
+        String choiceColumnHeader = headerData[columnIndex];
+        String[] choiceFields = choiceColumnHeader.split(":");
+        // filter by contest
+        String contestName = choiceFields[RcvChoiceHeaderField.CONTEST_NAME.ordinal()];
+        if (!contestName.equals(this.contestConfig.getContestId())) {
+          continue;
+        }
+        // validate and store the ranking associated with this choice column
+        String choiceName = choiceFields[RcvChoiceHeaderField.CHOICE_NAME.ordinal()];
+        if (!this.contestConfig.getCandidateCodeList().contains(choiceName)) {
+          Logger.log(Level.SEVERE, "Candidate: %s from cast vote record not found in config!", choiceName);
+          throw new CvrParseException();
+        }
+        Integer rank = Integer.parseInt(choiceFields[RcvChoiceHeaderField.RANK.ordinal()]);
+        columnIndexToRanking.put(columnIndex, new Pair<>(rank, choiceName));
+      }
+      // read all remaining rows and create CastVoteRecords for each one
+      while (true) {
+        String row = csvReader.readLine();
+        if (row == null) {
+          break;
+        }
+        // parse rankings
+        String[] cvrData = row.split(",");
+        ArrayList<Pair<Integer, String>> rankings = new ArrayList<>();
+        for (int columnIndex = CvrColumnField.ChoicesBegin.ordinal(); columnIndex < cvrData.length; columnIndex++) {
+          if (Integer.parseInt(cvrData[columnIndex]) == 1) {
+            // user marked this column
+            Pair<Integer, String> ranking = columnIndexToRanking.get(columnIndex);
+            rankings.add(ranking);
+          }
+        }
+        // create the cast vote record
+        CastVoteRecord castVoteRecord = new CastVoteRecord(this.contestConfig.getContestId(),
```

Comment:
Nit: There's a comment in CastVoteRecord that says:
```
// Dominion and Hart use this constructor
```
Please update this or probably better yet delete it (and the one about CDF and ES&S on the constructor below it).